h/api/schemas.py

```diff
@@ -264,17 +192,50 @@ class CreateAnnotationSchema(object):
     """Validate the POSTed data of a create annotation request."""
 
     def __init__(self, request):
-        self.structure = AnnotationSchema(request)
+        self.structure = AnnotationSchema()
+        self.request = request
 
     def validate(self, data):
-        return self.structure.validate(data)
+        appstruct = self.structure.validate(data)
+
+        new_appstruct = {}
+
+        _remove_protected_fields(appstruct)
+
+        new_appstruct['userid'] = self.request.authenticated_userid
+        new_appstruct['target_uri'] = appstruct.pop('uri', u'')
```

Comment:
So what should we do in this case? Is it valid to have an annotation with no URI? Or should we reject this with a validation error? Rather than trying to validate that it's an actual valid URI, we could for now at least check that it's a non-empty string but silly things like "a" would still be allowed